import java.util.*;

public class Week01and02Problems {
    public static void main(String[] args){

        int[] transactions = {500,300,200};

        findTwoSum(transactions,500);
    }

    static void findTwoSum(int[] nums,int target){

        HashMap<Integer,Integer> map = new HashMap<>();

        for(int i=0;i<nums.length;i++){

            int complement = target - nums[i];

            if(map.containsKey(complement)){
                System.out.println("Pair: "+complement+" + "+nums[i]);
                return;
            }

            map.put(nums[i],i);
        }
    }
}
